### PR TITLE
Fix NPE warning on `SconeSessionHandlerService`

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java [hunk 1]
```diff
@@ -54,20 +54,19 @@ public SconeSessionHandlerService(SconeSessionMakerService sessionService,
...
-                && teeSessionLogConfiguration.isDisplayDebugSessionEnabled()) {
+        if (teeSessionLogConfiguration.isDisplayDebugSessionEnabled()) {
             log.info("Session content [taskId:{}]\n{}",
-                    request.getTaskDescription().getChainTaskId(), session.toString());
+                    request.getTaskDescription().getChainTaskId(), session);
         }
         ResponseEntity<String> postSession = apiClient.postSession(session.toString());
-        int httpCode = postSession != null ? postSession.getStatusCodeValue() : null;
-        if (httpCode != 201) {
+        Integer httpCode = postSession != null ? postSession.getStatusCodeValue() : null;
```
Comment (jbern0rd) [inline src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java:68]: Did it happen ? `HttpStatus` is not marked as `Nullable` in `ResponseEntity`.
I am not sure it has a meaning to define a null `httpCode`.

**Review by jbern0rd — APPROVED** [review]

File: src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java (hunk 1 above)
Comment (mcornaton) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java:68]: I'm not sure of the range of values `postSession.getStatusCodeValue()` can have. Thus, I don't want to accidentally hide a Feign status code (do 0 or -1 have a special meaning, for example?). The easiest way is then to use an Integer and set its value to `null` when `postSession` is null.

Comment (jbern0rd) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java:68]: Wouldn't it be better to set `httpCode` to a `5xx` value for an internal server as we didn't get a response ?

Comment (mcornaton) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java:68]: What if we override a legit value? I'd prefer to be able to differentiate whether the CAS had a known issue - i.e. with a valid HTTP code - or if the workflow encountered another, unknown, issue - i.e. with a `null` value. 
As we use this value only for logging/debugging, I don't think using a `null` value - i.e. a non-valid value for an `HttpStatus` - would be an issue. Don't you think so?

Comment (jbern0rd) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java:68]: I don't think the case will ever happen due to how web standards have been defined and implemented.
If a connection is broken, it must be detected client-side and the HttpStatus will be returned as such.

It brings us to how the spring framework has implemented JSR305 (how to handle non-null related annotations).
They use it in a totally different way. By default, nothing can be null and the API must explicitly declare parameters which can be null. In ResponseEntity, http headers and status can never be null.

Could you add a specific log message when http status is null ? I would be really interested to see if the event ever happen in the future.

Comment (jbern0rd) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java]: No real need for a new logging event, we should see it from the `TeeController`.

Comment (mcornaton) [reply src/main/java/com/iexec/sms/tee/session/scone/SconeSessionHandlerService.java]: This is probably the best thing to do, you're right. Please see https://github.com/iExecBlockchainComputing/iexec-sms/pull/123/commits/720773f3e7433a688f7f5fe2b7d08a1efbfc572b :)

**Pull request merged**